### DISCUSSION TITLE
Smart Meter Interface (Iskra MT 174)

### DISCUSSION THREAD

**Discussion by kk98kk** [arendst/Tasmota#18725] [Q&A] [answered]
In advance I am new to working with microcontroller and Tasmota, so please be patient with me. 
I have an ESP32 and want to use the Smart Interface from Tasmota to send the energy data via MQTT to Home Assistant. My energy meter is an "Iskra MT 174". I added the following code in the user_config_override.h-file as you see in the picture.

![image](https://github.com/arendst/Tasmota/assets/88677709/aedd2838-9171-43e3-8bf2-d33b59ce2805)

In the platform_override.ini-file I only changed the device from tasmota to tasmota32 and used the right upload port.
After a successful build and upload to my ESP32, I could connect to my WiFi. I let the modul configuration for the GPIO-pins on "NONE" and added under Console->Edit Script following things: 
![image](https://github.com/arendst/Tasmota/assets/88677709/a9695352-0611-4307-be6b-89ce67cb1a5d)

Now it looks like this, but I get no data and I don't know why.
![image](https://github.com/arendst/Tasmota/assets/88677709/718ef725-c384-461a-950d-6292ea6311e5)

My Output from the consol looks like this:
![image](https://github.com/arendst/Tasmota/assets/88677709/09e82cb3-d49e-4258-8626-950837647905)

For test purpose I used my PI and used the following code and got this results: 
![image](https://github.com/arendst/Tasmota/assets/88677709/e5174b3b-6b6e-4b7d-b236-7daf851e66f8)

I am using the Version 12.5.0 from Tasmota

_Originally posted by @kk98kk in https://github.com/arendst/Tasmota/discussions/18721_

**Comment by Noschvie** [comment]
Have you tried to swap RX / TX ? Or check the coinnection using the SML debug mode (sensor53 dm = sets dump mode for meter m (1...N), must be set to 0 for normal operation)

**Reply by kk98kk** [reply, marked as answer]
Thank you for information to write in the console: sensor53 d1. I got the right values. After I changed script from 

```
>D
>B
->sensor53 r
>M 1
+1,3,o,0,300,STROM,1,100,2F3F210D0A
1,1-0:1.8.1*255(@1,Total Consumed,KWh,Total_in,3
1,1-0:2.8.1*255(@1,Total Delivered,KWh,Total_out,3
1,1-0:0.0.0*255(@#),Meter Number,,Meter_number,0
#
```

to this:
```
>D
>B
->sensor53 r
>M 1
+1,3,o,0,300,STROM,1,100,2F3F210D0A
1,1-0:1.8.0*255(@1,Total Consumed,KWh,Total_in,3
1,1-0:2.8.0*255(@1,Total Delivered,KWh,Total_out,3
1,1-0:16.7.0*255(@1,Current Consumption,KWh,Power_cur,3
1,0-0:C.1.0*255(@#),Meter Number,,Meter_number,0
#
```
it worked perfect for me.
I don´t know if only my Iskra MT174 has this protocoll or everyone. It could also be wrong in the documentation for the Iskra MT174.

**Comment by Jason2866** [comment]
Looking in the docs, there is a script for the MT-175. It has different serial speed and GPIOs are different.
```
>D
>B
->sensor53 r
>M 1
+1,3,s,16,9600,MT175
1,77070100010800ff@1000,E_in,kWh,E_in,1
1,77070100020800ff@1000,E_out,kWh,E_out,1
1,77070100100700ff@1,P,W,P,18
1,77070100240700ff@1,L1,W,L1,18
1,77070100380700ff@1,L2,W,L2,18
1,770701004C0700ff@1,L3,W,L3,18
1,77070100000009ff@#,Server_ID,,Server_ID,0
#
```
Maybe you try with this settings.

**Reply by kk98kk** [reply]
I tried it without success.

**Comment by nyunity** [comment]
Hi all, hi @kk98kk,

I tried your config and work, it shows the right total consumed. But the current consuption is empty. Did you change any other configs?

Happy to hear from you soon.

**Reply by kk98kk** [reply]
Hello, 

No. I didn´t change anything else. I just used the sript, which I posted here. 

Maybe your Smart Meter uses a different reference number for the current consumption or doesn´t support it at all. On my Smart Meter it was set to "1-0:16.7.0*255". You should be able to look up all your supported reference numbers on your Smart Meter by yourself and change it to the right and needed ones.

**Reply by nyunity** [reply]
@kk98kk did you insert a pin? Or worked it wirhout a PIN?

**Reply by kk98kk** [reply]
No. It worked without a PIN.

**Reply by nyunity** [reply]
damn, can you tell me where I can find these reference numbers? and, did you change the config to module type generic 18 or default Sonoff Basic (1)?

**Reply by kk98kk** [reply]
I don´t have that Smart Meter anymore, but from pictures from the internet it should be right there:

![image](https://github.com/arendst/Tasmota/assets/88677709/08c72529-bf16-4c7e-849e-9571e0669b04)





I don´t know what you mean by that. My config module looks like this:

![image](https://github.com/arendst/Tasmota/assets/88677709/07814976-3077-48bc-85a1-abdce4cd1a08)

**Reply by nyunity** [reply]
okay, i think i crashed my hichii. I upgrade my fw ti 12.5.0 and now I am not able no edit config anymore..

**Reply by hagenbuch** [reply]
If you still are able to access the Hichi, you should just be able to upload a correct firmware (from Hichi..)
